RHDEVDOCS-5014: Release Notes updates for Pipelines 1.9.1 (#56267)
openshift/openshift-docs: modules/op-release-notes-1-9.adoc

// .PAC	

> * Before this update, {pac} did not get updated values from the {pac} ConfigMap object. With this update, this issue is fixed and {pac} ConfigMap object looks for any new changes.

Author: Fixed.

* Before this update, a pull request failed if the user in the annotation provided values by using a regex form, for example, `refs/head/rel-*`. The pull request failed because it was missing `refs/heads` in its base branch. With this update, the prefix is added and checked that it matches. This resolves the issue and the pull request is successful.	
// Savita Ashture

[id="release-notes-1-9-1_{context}"]
== Release notes for {pipelines-title} General Availability 1.9.1

> With this update, {pipelines-title} General Availability (GA) 1.9.1 is available on {product-title} 4.12.

Reviewer: It is available also on 4.11

Author: Fixed.

> 

Reviewer 2: We need to add a known issue section to add this issue https://issues.redhat.com/browse/SRVKP-2854

The value for `CATALOG_REFRESH_INTERVAL`  provided by the user in Tekton Hub CR is not getting updated in the Hub API ConfigMap and the default value of 30m is used by Tekton Hub

Author: Hey @PuneetPunamiya - Could you also tell me if we have a workaround for this user too? In the known issues section, we advise the user by giving a workaround as well, if possible.

Reviewer 3: @Dhruv-Soni11 we do not need any workaround here because the default value is 30m so the API will refresh after 30m , is that correct @PuneetPunamiya

Author: Thanks for letting me know. I have made the changes as you suggested @rupalibehera .

Reviewer 2: - Workaround
  - update `tekton-hub-api` Configmap from the targetNamespace manually with your changes i.e. update the value for `CATALOG_REFRESH_INTERVAL` and delete the existing `tekton-hub-api-xxx` pod from the targetNamespace

Reviewer 2: Also we need to add a note that

With any update/upgrade the config map will be updated using default values

Reviewer 3: @Dhruv-Soni11 lets merge the PR we can add this info in future releases

[id="fixed-issues-1-9-1_{context}"]	
== Fixed issues	

> * Before this update, the `tkn pac repo list` command did not run on Microsoft Windows. With this update, this issue is resolved and now you can run the `tkn pac repo list` command on on Microsoft Windows.

Author: Fixed.

> * Before this update, {pac} watcher did not receive all of the configuration change events. With this update, this issue is resolved. The {pac} watcher is updated and does not miss the configuration change events.

Author: Fixed.

> * Before this update, TaskRuns or PipelineRuns pods created by {pac} could not access custom certificates exposed by the user in the cluster. With this update, this issue has been resolved. You can now access custom certificates in the cluster from TaskRuns or PipelineRuns pods.

Author: Fixed.

> * Before this update, the core interceptor `tekton-triggers-core-interceptors` used in the `Trigger` resource did not function after the Pipelines operator was upgraded to version 1.9. With this update, this issue is resolved. The core interceptor, `tekton-triggers-core-interceptors`, has been updated to TLS version 1.2, which ensures that the functionality of the core interceptor runs accurately.

Reviewer 4: How about changing to something like this because I want to mention that this issue occurred on a Openshift cluster where FIPS enabled
Please feel free to modify sentence but make sure to mention FIPS
```suggestion
* Before this update, the core interceptor `tekton-triggers-core-interceptors` used in the `Trigger` resource did not function after the Pipelines operator was upgraded to version 1.9 on a cluster where FIPS enabled. With this update, this issue is resolved. The core interceptor, `tekton-triggers-core-interceptors`, has been updated to TLS version 1.2 (because OpenShift uses MInTLS as 1.2 for all components), which ensures that the functionality of the core interceptor runs accurately.
```

Author: Fixed.

> * Before this update, when using a pipeline run with an internal OpenShift image registry, the URL to the image was hardcoded in the pipeline run definition. For example:

Reviewer: IMHO `was hardcoded` should be `had to be hardcoded`... this is a very specific use-case, customers might not need it at all.

Author: Fixed.

+
[source,yaml]
----
...
  - name: IMAGE_NAME
    value: 'image-registry.openshift-image-registry.svc:5000/<test_namespace>/<test_pipelinerun>'
...
----
+
When using in the context of {pac}, such hardcoded values prevents pipeline run definitions from being used in different clusters and namespaces.
+
With this update, you can generalize the pipeline run definitions by using the dynamic template variables, instead of hardcoding the values for namespaces and pipeline run names. For example:
+
[source,yaml]
----
...
  - name: IMAGE_NAME

>     value: 'image-registry.openshift-image-registry.svc:5000/{{ namespace }}/{{ pipeline_run_name }}'

Reviewer: 1. it should be `{{ target_namespace }}`
2. pipeline run name was not implemented, one can use `$(context.pipelineRun.name)`

Author: Fixed.

...
----

> * With this update, {pac} uses the same GitHub token to fetch a remote task available in the same host on GitHub.

Reviewer: This is related to this 1.9.0 RN item 

```
Before this update, the GitHub apps token scoping was not defined and tokens would be used on every repository installation. With this update, you can scope the GitHub apps token to the target repository using the following parameters:

secret-github-app-token-scoped: Scopes the app token to the target repository, not to every repository the app installation has access to.

secret-github-app-scope-extra-repos: Customizes the scoping of the app token with an additional owner or repository.
```

In 1.9.0 it worked only with default git branch, now it works with any branch

Author: Fixed.

>     value: 'image-registry.openshift-image-registry.svc:5000/{{ target_namespace }}/{{ $(context.pipelineRun.name) }}'

Reviewer: The correct format of string is `'image-registry.openshift-image-registry.svc:5000/{{ target_namespace }}/$(context.pipelineRun.name)'` (without curly braces for second variable)

Reviewer: @chmouel @savitaashture BTW, what was the motivation for implementing target_namespace (and not pipeline run name)? Namespace can be set using context variable so either both should be implemented in PAC or none.

Reviewer 5: @ppitonak the request was to be able to used anywhere in the pipelinerun, context params cannot be everywhere,

> * Before this update, {pac} used the same GitHub token to fetch a remote task available in the same host on the default GitHub branch. With this update, this issue is now resolved. {pac} uses the same GitHub token to fetch a remote task from any GitHub branch.

Reviewer: I would add word `only` (on the default GitHub branch) to the first sentence so that second sentence makes sense.

// Savita Ashture

[id="release-notes-1-9-1_{context}"]
== Release notes for {pipelines-title} General Availability 1.9.1

With this update, {pipelines-title} General Availability (GA) 1.9.1 is available on {product-title} 4.11 and later.

[id="fixed-issues-1-9-1_{context}"]	
== Fixed issues	

* Before this update, the `tkn pac repo list` command did not run on Microsoft Windows. With this update, this issue is resolved and now you can run the `tkn pac repo list` command on on Microsoft Windows.

* Before this update, {pac} watcher did not receive all of the configuration change events. With this update, this issue is resolved. The {pac} watcher is updated and does not miss the configuration change events.

* Before this update, TaskRuns or PipelineRuns pods created by {pac} could not access custom certificates exposed by the user in the cluster. With this update, this issue has been resolved. You can now access custom certificates in the cluster from TaskRuns or PipelineRuns pods.

> * Before this update, the core interceptor, `tekton-triggers-core-interceptors`, used in the `Trigger` resource did not function after the Pipelines operator was upgraded to version 1.9, on a cluster which is enabled with FIPS. With this update, this issue is resolved. The core interceptor, `tekton-triggers-core-interceptors`, has been updated to TLS version 1.2, (because OpenShift uses MInTLS 1.2 for all it's components), which ensures that the functionality of the core interceptor runs accurately.

Reviewer 6: ```suggestion
* Before this update, on a cluster enabled with FIPS, the `tekton-triggers-core-interceptors` core interceptor used in the `Trigger` resource did not function after the Pipelines Operator was upgraded to version 1.9. This update resolves the issue. Now, OpenShift uses MInTLS 1.2 for all its components. As a result, the `tekton-triggers-core-interceptors` core interceptor updates to TLS version 1.2and its functionality runs accurately.
```

Author: Fixed.

* Before this update, when using a pipeline run with an internal OpenShift image registry, the URL to the image had to be hardcoded in the pipeline run definition. For example:
+
[source,yaml]
----
...
  - name: IMAGE_NAME
    value: 'image-registry.openshift-image-registry.svc:5000/<test_namespace>/<test_pipelinerun>'
...
----
+

> When using in the context of {pac}, such hardcoded values prevents pipeline run definitions from being used in different clusters and namespaces.

Reviewer 6: ```suggestion
When using a pipeline run in the context of {pac}, such hardcoded values prevented the pipeline run definitions from being used in different clusters and namespaces.
```

Author: Fixed.

+

> With this update, you can generalize pipeline run definitions by using the dynamic template variables, instead of hardcoding the values for namespaces and pipeline run names. For example:

Reviewer 6: ```suggestion
With this update, you can use the dynamic template variables instead of hardcoding the values for namespaces and pipeline run names to generalize pipeline run definitions. For example:
```

Author: Fixed.

> With this update, you can generalize the pipeline run definitions by using the dynamic template variables, instead of hardcoding the values for namespaces and pipeline run names. For example:

Reviewer 6: ```suggestion
With this update, you can use the dynamic template variables instead of hardcoding the values for namespaces and pipeline run names to generalize the pipeline run definitions. For example:
```

**Note:** @Dhruv-Soni11 I am not sure about the usage of the term: pipeline run names
Please check with the SMEs and refer to the existing docs to verify this usage and way of writing.

+
[source,yaml]
----
...
  - name: IMAGE_NAME
    value: 'image-registry.openshift-image-registry.svc:5000/{{ target_namespace }}/$(context.pipelineRun.name)'
...
----

> * Before this update, {pac} used the same GitHub token to fetch a remote task available in the same host only on the default GitHub branch. With this update, this issue is now resolved. {pac} uses the same GitHub token to fetch a remote task from any GitHub branch.

Reviewer 6: ```suggestion
* Before this update, {pac} used the same GitHub token to fetch a remote task available in the same host only on the default GitHub branch. This update resolves the issue. Now {pac} uses the same GitHub token to fetch a remote task from any GitHub branch.
```

Author: Fixed.

[id="known-issues-1-9-1_{context}"]
=== Known issues

> * The value for `CATALOG_REFRESH_INTERVAL`, a field in the Hub API ConfigMap used in the Tekton Hub CR, does not get updated with a custom value provided by the user.

Reviewer 6: ```suggestion
* The value for `CATALOG_REFRESH_INTERVAL`, a field in the Hub API `ConfigMap` object used in the Tekton Hub CR, is not getting updated with a custom value provided by the user.
```

Author: Fixed.
